File: backend/supabase_client.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv()

def get_supabase_client() -> Client:
    """
    Initializes and returns a Supabase client.
    Ensure SUPABASE_URL and SUPABASE_ANON_KEY are set in your .env file.
    """
    url: str = os.environ.get("SUPABASE_URL")
    key: str = os.environ.get("SUPABASE_ANON_KEY")
    
    if not url or not key:
        raise ValueError("SUPABASE_URL or SUPABASE_ANON_KEY not found in environment variables.")
        
    return create_client(url, key)

# Single instance of the client
supabase: Client = get_supabase_client()

File: frontend/src/supabaseClient.js
import { createClient } from '@supabase/supabase-js'

const supabaseUrl = 'https://knqeiskbjdeyzaogdjhp.supabase.co'
const supabaseAnonKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImtucWVpc2tiamRleXphb2dkamhwIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzcwMjY4ODMsImV4cCI6MjA5MjYwMjg4M30.hzLlj9FYgZZPd_vvZ5KOIx2ZMf2d1DHSIUlOOAZ0G-o'

export const supabase = createClient(supabaseUrl, supabaseAnonKey)
